addons/account/models/account_tax_report.py

```diff
@@ -260,3 +271,22 @@ def _validate_tags(self):
 
             if neg_tags.name != '-'+record.tag_name or pos_tags.name != '+'+record.tag_name:
                 raise ValidationError(_("The tags linked to a tax report line should always match its tag name."))
+
+    def is_carry_over(self, options, line_amount, carried_over_amount):
+        """
+        Override this method to set the condition for a line to be carried over to the next period
+        :param options: The options of the reports
+        :param line_amount: The amount on the line
+        :param carried_over_amount: The amount carried over by the analytic account for this line
+        :return: True if this line is
+        """
+        self.ensure_one()
+        # Carry over is disabled by default, but if there is a carry over condition  method on the line we are
+        # calling it first. That way we can have a default carryover condition for the whole report (is_carry_over)
+        # and specialized condition for specific lines if needed
+        if self.carried_over_condition_method:
+            condition_method = getattr(AccountTaxReport, self.carried_over_condition_method, False)
+            if condition_method:
+                return condition_method(options, line_amount, carried_over_amount)
+
+        return False
```

Comment:
Actually, we could tweak that yet a little more. The function could return a tuple of two elements instead of True (and None otherwise): the upper and lower bounds to carry over for. This way, we wouldn't only carry over negative lines, but would have a fully customizable framework; and I'm sure l10n modules will love that :) And if there is no upper or lower bound, just leave None in the tuple. So, this also changes what needs to be returned by carried_over_condition_method.